sklearn/tests/test_naive_bayes.py

```diff
@@ -513,18 +514,20 @@ def test_mnb_prior_unobserved_targets():
 
     clf = MultinomialNB()
 
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
```

Comment:
From the commit that added this test: https://github.com/scikit-learn/scikit-learn/commit/0682d92aeb48d3acb7b78174afc35e97b47b948d, I think this is a `RuntimeWarning`:

```suggestion
        warnings.simplefilter("error", RuntimeWarning)
```

Reply:
Hi @thomasjpfan, if I understand correctly that warning is now handled at https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/naive_bayes.py#L571 and is no longer raised to the user.
I think the only warning that is being raised by the naive bayes code is the UserWarning at https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/naive_bayes.py#L591.
if my understanding is correct, maybe we could drop the warning check altogether then?

Reply:
We can not drop the test here. This test is checking that:

https://github.com/scikit-learn/scikit-learn/blob/41b8f84f1f1eb5930c0ec60598ca1748005a1f75/sklearn/naive_bayes.py?rgh-link-date=2022-04-06T20%3A12%3A59Z#L568-L572 

works as expected. In other words, the implementation is catching the warning and a user does not see the runtime warning at all.

The alpha warning can not be trigger here since alpha is 1.0 by default.

Reply:
Ah, of course. That makes sense, thanks for explaining! I'll adapt the code.